nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -178,6 +178,12 @@ void RegulatedPurePursuitController::configure(
   global_path_pub_ = node->create_publisher<nav_msgs::msg::Path>("received_global_plan", 1);
   carrot_pub_ = node->create_publisher<geometry_msgs::msg::PointStamped>("lookahead_point", 1);
   carrot_arc_pub_ = node->create_publisher<nav_msgs::msg::Path>("lookahead_collision_arc", 1);
+
+  // initialize collision checker and set costmap
+  collision_checker_ = std::make_unique<nav2_costmap_2d::
+      FootprintCollisionChecker<nav2_costmap_2d::Costmap2D *>>(costmap_);
+  collision_checker_->setCostmap(costmap_);
+  traverse_unknown = true;
```

Comment:
Since this is a class variable, needs a trailing underscore. Also populate it from `costmap_ros_->getLayeredCostmap()->isTrackingUnknown())` rather than setting it to always true

Reply:
setting the variable to this function in the configure method fails basic api tests. so i removed the variable and get the value  directly in incollision function